lychee-lib/src/collector.rs

```diff
@@ -13,15 +13,17 @@ use par_stream::ParStreamExt;
 pub struct Collector {
     base: Option<Base>,
     skip_missing_inputs: bool,
+    use_html5ever: bool,
 }
 
 impl Collector {
     /// Create a new collector with an empty cache
     #[must_use]
-    pub const fn new(base: Option<Base>, skip_missing_inputs: bool) -> Self {
+    pub const fn new(base: Option<Base>, skip_missing_inputs: bool, use_html5ever: bool) -> Self {
```

Comment:
beautiful function signature of a bunch of bools. probably refactor it to use options struct? or just make all fields pub?

Reply:
you can add a new field for Collector

Reply:
you gave me an idea, i'm just gonna make the entire thing a builder pattern and be done

Reply:
there's no need for builder pattern here...

Reply:
just make the new fields public and it would be sufficient.

Reply:
in absence of named arguments and default arguments i think what I pushed is preferrable but i'll leave the decision to somebody else

Reply:
>just make the new fields public and it would be sufficient.

you mean because we're gonna remove the field again?

Reply:
nope.
There's not much gain for builder pattern here when you can explicit specify a field when constructing a Collector.
The added function just adds unnecessary function calls.

Reply:
I don't understand. By exposing only a single field as pub without touching the constructor function, you end up with API like this:

```
let mut c = Collector::new(x, false);
c.use_html5ever = true;
```

is that the proposal?

Reply:
okay just keep your build pattern.
we will eventually remove them anyway...

Reply:
sorry, I didn't mean to come off as too opinionated here. I'm happy to do it whichever way y'all think is best.

> we will eventually remove them anyway...

yeah that's why I was wondering, because your solution does keep the constructor signature the same

Reply:
Your builder pattern would look better than
```rust
let mut c = Collector::new(x, false);
c.use_html5ever = true;
```
And in the long run we will remove the `use_html5ever` field eventually.
So just use your design for now.